wrapspawner/wrapspawner.py

```diff
@@ -220,6 +219,56 @@ def clear_state(self):
         super().clear_state()
         self.child_profile = ''
 
+class FilteredSpawner(ProfilesSpawner):
+
+    """ProfilesSpawner - leverages the Spawner options form feature to allow user-driven
+        configuration of Spawner classes while permitting:
+        1) configuration of Spawner classes that don't natively implement options_form
+        2) administrator control of allowed configuration changes
+        3) runtime choice of which Spawner backend to launch
+    """
+    default_profiles = List(
+        trait = Tuple( Unicode(), Unicode(), Type(Spawner), Dict(), Unicode() ),
+        default_value = [],
+        config = True,
+        help = """List of profiles to offer in addition to docker images for selection. Signature is:
+                List(Tuple( Unicode, Unicode, Type(Spawner), Dict , Unicode)) corresponding to
+                profile display name, unique key, Spawner class, dictionary of spawner config options, comma separated list of authorized groups.
+                The first three values will be exposed in the input_template as {display}, {key}, and {type}"""
+        )
+    
+    def get_user_groups(self):
+        import subprocess, sys
+        user = self.user.name
+        cmd_result = subprocess.Popen("groups "+user, shell=True, stdout=subprocess.PIPE).stdout.read()
+        #Get system default encoding
+        sys_encoding = sys.getdefaultencoding()
+        groups_list = cmd_result.decode(sys_encoding).split(':')[1].replace('\n', '').strip().split(' ')
+        return groups_list
+    
+    def _user_profiles(self):
```

Comment:
Rather than implementing logic specific to your use case (filtering based on OS groups), maybe this filter should itself be a hook and then your configuration would register the appropriate filter function for your use case?